lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -2132,6 +2112,35 @@ def each_host_range_chunk(host_ranges, &block)
 
   #######
   private
+
+  def run_nmap(arguments, nmap, use_sudo=false)
+    begin
+      nmap_pipe = use_sudo ? ::Open3::popen3('sudo', nmap, *arguments) : ::Open3::popen3([nmap, 'nmap'], *arguments)
```

Comment:
Looks like the argv0 (the name of the program) was previously specified when an nmap path was called with popen3, just confirming if we need to do anything similar in the `sudo` scenario?

> The parameters env, cmd, and opts are passed to Process.spawn.
> A commandline string and a list of argument strings can be accepted as follows:
>
>     Open3.popen3("echo abc") {|i, o, e, t| ... }
>     Open3.popen3("echo", "abc") {|i, o, e, t| ... }
>     Open3.popen3(["echo", "argv0"], "abc") {|i, o, e, t| ... }
>
> https://github.com/ruby/ruby/blob/903af74bbd2998cbc159c70010fdba0543509a0f/lib/open3.rb#L58-L63

Reply:
I can't think of any reason why we'd need to add it explicitly...but I can?

Reply:
I'd be in favor of removing it entirely if it makes sense to do so.

Based on the nmap code it assumes that program_name is a valid path from what I can tell:
https://github.com/nmap/nmap/search?q=argv0

I tried going down the git blame route but there wasn't much detail about why the name was specified like this
https://github.com/rapid7/metasploit-framework/commit/0cc3db65bec12a2acb8167205c95f12e732b9856

I'm thinking the original intent was to fix a weird cywin issue? It looks like the parent commit tried to unshift `nmap` as the program name which didn't work:
https://github.com/rapid7/metasploit-framework/commit/dc2c3557038c545047a8517857c4fbfa18c942a5